test/numeric_literal_separators.coffee

```diff
@@ -0,0 +1,33 @@
+
+test 'numeric literal separators', ->
+  compiles = (code) -> doesNotThrow ->
+    CoffeeScript.compile code, bare: yes
+  doesNotCompile = (code) -> throws ->
+    CoffeeScript.compile code, bare: yes
+
+  compiles '1_2.34_5e6_7'
+  doesNotCompile '1_.23'
+  doesNotCompile '1._23'
+  doesNotCompile '1e_2'
+  doesNotCompile '1e2_'
+  doesNotCompile '1_'
+  doesNotCompile '1__2'
+  compiles '0x1_2_3_4'
+  doesNotCompile '0x_1234'
+  doesNotCompile '0x1234_'
+  doesNotCompile '0x1__34'
+  compiles '0b10_10'
+  doesNotCompile '0b_100'
+  doesNotCompile '0b100_'
+  doesNotCompile '0b1__1'
+  compiles '0o7_7_7'
+  doesNotCompile '0o_777'
+  doesNotCompile '0o777_'
+  doesNotCompile '0o6__6'
+
+  eq 123456, 123_456
+  eq 10**11, 1_0e1_0
+  eq 12.345, 1_2.34_5
+  eq 0x1234, 0x12_34
+  eq 0b1000, 0b10_00
+  eq 0o777, 0o7_7_7
```

Comment:
Can you please separate these into a few separate tests? We don’t need every `eq` and `compiles` and `doesNotCompile` line to be its own test, but it’s a little much to group them all into one huge test function. It’s easier to debug things in the future if we can see that say the octal numeric literal separators tests failed, but not the other numeric literal separators tests.

Reply:
Done. Hope it's fine like this.

Reply:
This looks good. Unfortunately we have a failure in CI: https://travis-ci.org/jashkenas/coffeescript/jobs/593738497

```
failed 1 and passed 1307 tests in 7.79 seconds 
  invalid decimal literal separators do not compile 
  AssertionError [ERR_ASSERTION]: Missing expected exception.
    at doesNotCompile (/home/travis/build/jashkenas/coffeescript/test/numeric_literal_separators.coffee:1:1)
    at Function.<anonymous> (/home/travis/build/jashkenas/coffeescript/test/numeric_literal_separators.coffee:1:1)
    at global.test (/home/travis/build/jashkenas/coffeescript/Cakefile:556:21)
    at Object.<anonymous> (/home/travis/build/jashkenas/coffeescript/test/numeric_literal_separators.coffee:1:1)
    at Object.<anonymous> (/home/travis/build/jashkenas/coffeescript/test/numeric_literal_separators.coffee:1:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.CoffeeScript.run (/home/travis/build/jashkenas/coffeescript/lib/coffeescript/index.js:67:23)
    at runTests (/home/travis/build/jashkenas/coffeescript/Cakefile:628:22)
    at Object.action (/home/travis/build/jashkenas/coffeescript/Cakefile:642:12)
    at invoke (/home/travis/build/jashkenas/coffeescript/lib/coffeescript/cake.js:57:26)
    at Object.exports.run (/home/travis/build/jashkenas/coffeescript/lib/coffeescript/cake.js:87:20)
    at Object.<anonymous> (/home/travis/build/jashkenas/coffeescript/bin/cake:22:42)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
 
  function() {
    doesNotCompile('1_.23');
    doesNotCompile('1._23');
    doesNotCompile('1e_2');
    doesNotCompile('1e2_');
    doesNotCompile('1_');
    return doesNotCompile('1__2');
  }
```

Reply:
That's odd. The test passes on my machine. Ideas?

Reply:
Just guessing from afar, but perhaps it's from the package-lock.json reversion?

Reply:
Passed before and after that.